### PR TITLE
FIX-2093 Edit Curves

### LINKED ISSUE #2093: Ability to easily change the set of selected curves in a Log💡
Today when I have selected 10 curves in a log, and I want to change the selection to for instance 12 curves (deselecting some curves and adding others), this is not easy. I typically have to start all over. I also want Wex to remember the interval I have set when displaying the Logs.

**Describe the solution you'd like**
I would like a "edit curve" button, which brings up a list of all the curves in the log, where I see the currently selected curves, and I am able to change the selection (add, remove).

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2093 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

Added the possibility to change the selected mnemonics while in CurveValuesView.

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx [hunk 1]
```diff
@@ -1,52 +1,87 @@
...
 
-interface EditIntervalProps {
+interface EditSelectedLCIProps {
   disabled?: boolean;
   overrideStartIndex?: string;
   overrideEndIndex?: string;
 }
 
-const EditInterval = (props: EditIntervalProps): React.ReactElement => {
+const EditSelectedLCI = (props: EditSelectedLCIProps): React.ReactElement => {
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx]: Abbreviations like LCI can be confusing for new developers. I suggest changing it to `EditSelectedLogCurveInfo`.

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx [hunk 2]
```diff
@@ -1,52 +1,87 @@
...
+    }
+  }, [selectedLog]);
+
   useEffect(() => {
     if (overrideStartIndex) setStartIndex(getParsedValue(overrideStartIndex));
     if (overrideEndIndex) setEndIndex(getParsedValue(overrideEndIndex));
   }, [overrideStartIndex, overrideEndIndex]);
 
-  const submitEditInterval = () => {
+  const submitLCI = () => {
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx]: `submitLCI` -> `submitLogCurveInfo`

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx [hunk 3]
```diff
@@ -1,52 +1,87 @@
...
     setIsEdited(false);
-    const logCurveInfoWithUpdatedIndex = selectedLogCurveInfo.map((logCurveInfo) => {
+    const filteredLCI = logCurveInfo.filter((lci) => selectedMnemonics.includes(lci.mnemonic));
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx]: `filteredLCI` -> `filteredLogCurveInfo`

**Review by janmarius — CHANGES_REQUESTED** [review]

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx (hunk 1 above)
Comment (eliasbruvik) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLCI.tsx]: Removed all abbreviations except for in arrow functions.

**Review by janmarius — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**